.github/workflows/CI.yml

```diff
@@ -0,0 +1,26 @@
+name: CI
+
+on: 
+  pull_request:
+    types:
+      - opened
+      - edited
+      - synchronize
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  Run-Tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          enable-cache: true
+
+      - name: Run Tests
+        run: uv run pytest
```

Comment:
Add linting, similar to [https://github.com/davidbrownell/UvRepoTemplate/blob/d969b0404194f48659e7c0991e56f54c052e5919/.github/workflows/CICD_impl.yml#L70](https://github.com/davidbrownell/UvRepoTemplate/blob/d969b0404194f48659e7c0991e56f54c052e5919/.github/workflows/CICD_impl.yml#L70)

(Run locally first to make sure all the linter errors are addressed)